tests/ui/async-await/pin-ergonomics/borrow.rs

```diff
@@ -0,0 +1,31 @@
+//@ check-pass
+
+#![feature(pin_ergonomics)]
+#![allow(dead_code, incomplete_features)]
+
+// Makes sure we can handle `&pin mut place` and `&pin const place` as sugar for
+// `unsafe { Pin::new_unchecked(&mut place) }` and `Pin::new(&place)`.
```

Comment:
This is wildly unsound, which gives me pause even about doing it on nightly, but the feature flag is marked as incomplete, and we're planning to follow-up with the borrow checker rules, so I suppose this is OK.

Reply:
I move the place in `&pin mut place` like `std;:pin::pin!(place)` to ensure soundness before we implement the borrowing rules of `&pin mut|const`.